### ISSUE TITLE
IMAGE command doesn't support .svg files

### ISSUE THREAD

**Issue by soracoder** [guigrpa/docx-templates#33]
Hi!
I have noticed that trying to insert an svg file in the word document with the command +++IMAGE insertSvg()+++ always ends with the document broken and an image that cannot be displayed

![image](https://user-images.githubusercontent.com/8900198/45294822-e4e87d00-b4fc-11e8-80bf-255c9f88acbf.png)

Is there any plan to support SVG files? or maybe I'm missing something?
I'm using the same function to load a .png file and everything goes well.

**Comment by guigrpa** [comment]
Are SVG files supported in Word?

**Comment by vdechef** [comment]
Seems like docx does not support SVG format. But SVG can be converted to EMF, and EMF can be imported.
See https://www.quora.com/How-would-you-insert-a-svg-file-in-PowerPoint

**Comment by soracoder** [comment]
Hi!
Yes, Word support .svg files without a problem.
It seems that it does not handle it as the other image files, it needs a structure like this:
`<a:graphic xmlns:a="http://schemas.openxmlformats.org/drawingml/2006/main">
              <a:graphicData uri="http://schemas.openxmlformats.org/drawingml/2006/picture">
                <pic:pic xmlns:pic="http://schemas.openxmlformats.org/drawingml/2006/picture">
                  <pic:nvPicPr>
                    <pic:cNvPr id="2" name="mct-logo.svg"/>
                    <pic:cNvPicPr/>
                  </pic:nvPicPr>
                  <pic:blipFill>
                    <a:blip r:embed="rId8">
                      <a:extLst>
                        <a:ext uri="{28A0092B-C50C-407E-A947-70E740481C1C}">
                          <a14:useLocalDpi xmlns:a14="http://schemas.microsoft.com/office/drawing/2010/main" val="0"/>
                        </a:ext>
                        <a:ext uri="{96DAC541-7B7A-43D3-8B79-37D633B846F1}">
                          <asvg:svgBlip xmlns:asvg="http://schemas.microsoft.com/office/drawing/2016/SVG/main" r:embed="rId9"/>
                        </a:ext>
                      </a:extLst>
                    </a:blip>
                    <a:stretch>
                      <a:fillRect/>
                    </a:stretch>
                  </pic:blipFill>
                  <pic:spPr>
                    <a:xfrm>
                      <a:off x="0" y="0"/>
                      <a:ext cx="1371600" cy="1371600"/>
                    </a:xfrm>
                    <a:prstGeom prst="rect">
                      <a:avLst/>
                    </a:prstGeom>
                  </pic:spPr>
                </pic:pic>
              </a:graphicData>
            </a:graphic>`

Notice the `<asvg:svgBlip xmlns:asvg="http://schemas.microsoft.com/office/drawing/2016/SVG/main" r:embed="rId9"/>` inside the `<a:ext>`
I'm preparing a PR for this.

**Comment by guigrpa** [comment]
Thanks, I look forward to it!

**Comment by soracoder** [comment]
Hi again.
I don't know why but when word imports an svg it references this:

```
<a:blip r:embed="rId4">
  <a:extLst>
    <a:ext uri="{96DAC541-7B7A-43D3-8B79-37D633B846F1}">
      <asvg:svgBlip xmlns:asvg="http://schemas.microsoft.com/office/drawing/2016/SVG/main" r:embed="rId5"/>
    </a:ext>
  </a:extLst>
</a:blip>
```
It has 2 r:embed arguments one in the a:blip tag and another in the asvg:svgblip.
The one in the asvg:svgblip is correspondant to the svg file and the other one seems to be referencing a png file that word auto creates when you import the svg (some king of preview)

The thing is, if you don't reference a bitmap image on the a:blip (whichever image, word only shows the svg file), word does not create the svg file. So it's needed some kind of predefined image to load svg files.

My main problem is that although I successfully create the document, when I'm going to open it, word sais that is corrupt and tries to fix it. It opens the file and the svg is loaded in, but is not good that word propts that it's a broken file.
I have taken a look at the fixed one and it seems to be no diference.

Any thoughts?

**Comment by lwallent** [comment]
Thanks for a great templating package!
Based on the additions in my Pull Request (#61) I have now successfully added SVG images to the reports I generate using the `docx-templates` module. SVG works really well for dynamically generated reports.
Please let me know if you see any problems with the created solution.
The` images-path.docx` example is now working as expected plus I have added an additional example called `images-svg.docx`

**Comment by jjhbw** [comment]
I've worked on @lwallent 's PR a bit and the result is 4f1df7000917bd0dc4e4a1283548a80462ba9da0 on the next branch, which will be part of the next release.

The thumbnail does indeed seem required for Word to render the SVG correctly. Weirdly, even modern versions of Word will flat out refuse to render the SVG if there is no reference to a thumbnail present in the xml. However, it seems that you also just provide an empty thumbnail.

In light of this, I have made providing a PNG thumbnail optional. If not provided, the library will simply inject a nonsense thumbnail consisting of a tiny base64 string. Modern versions of Word will simply ignore this anyway and render the SVG. I have updated the README to reflect this, but noted that providing a fallback thumbnail is good practice.

**Comment by jjhbw** [comment]
One thing i currently don't understand yet are the implications of providing a constant URI (`96DAC541-7B7A-43D3-8B79-37D633B846F1`) for the thumbnail's XML`Node`. It doesn't seem to be causing issues at this time, even when multiple thumbnails with this URI are inserted.

**Comment by jjhbw** [comment]
Will be included in the next release (4.1.0) which has been finalised today.
